app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/viewmodel/ListViewModel.kt

```diff
@@ -1,24 +1,30 @@
 package com.example.marcinwisniewski.intivetrainingapp.movielist.viewmodel
 
+import android.arch.lifecycle.LiveData
 import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.Transformations
 import android.arch.lifecycle.ViewModel
 import com.example.marcinwisniewski.intivetrainingapp.movielist.model.Movie
-import com.example.marcinwisniewski.intivetrainingapp.movielist.model.MovieProvider
+import com.example.marcinwisniewski.intivetrainingapp.movielist.model.MovieRepository
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
 import javax.inject.Inject
 
-class ListViewModel @Inject constructor(private val movieRepository: MovieProvider) : ViewModel() {
-    fun getMovies(): MutableLiveData<List<Movie>> {
-        // TODO: create handle error mechanism
-        val moviesList = MutableLiveData<List<Movie>>()
-        movieRepository.getMoviesResponse()
+class ListViewModel @Inject constructor(private val movieRepository: MovieRepository) : ViewModel() {
+
+    private val mutableMovieList: MutableLiveData<List<Movie>> = MutableLiveData()
+    val movieList: LiveData<List<Movie>> = Transformations.map(mutableMovieList) { mutableMovieList ->
```

Comment:
Something is wrong here. What is reason of this mapping ?

Reply:
I simplified it and i am only assigning mutable list to livedata now, without transformation- i had no other idea how to get the movieList immutable in other way. ( livedata have setValue method private but assigning mutableList to livedata list gives oportunity to fetch data and it is impossible to change it - it is going to be passed as an argument in custom binding adapter to avoid observing it in fragment.